hermes/crates/cbork/cddl-parser/tests/byte_sequences.rs

```diff
@@ -1,91 +1,80 @@
+    "abcdefghijklmnopq $  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\t",
+    "abcdefghijklmnopq %  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\n",
+    "abcdefghijklmnopq ^  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\r",
+    "abcdefghijklmnopq &  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\r\n",
+];
+
+pub(crate) const BYTES_PASSES: &[&str] = &[
+    "h''",
+    "b64''",
+    "''",
+    "h'00'",
+    "h'63666F6FF6'",
+    "h'68656c6c6f20776f726c64'",
+    "h'4 86 56c 6c6f'",
+    "h' 20776 f726c64'",
+    "h'00112233445566778899aabbccddeeff0123456789abcdef'",
+    "h'0 1 2 3 4 5 6 7 8 9 a b c d e f'",
+    "h' 0 1 2 3 4 5\r 6 7 \n 8 9 a\r\n\t b c d e f'",
+    "h'0 \n\n\r f'",
+    "b64'aHR0cHM6Ly93d3cuZXhhbXBsZS5jb20vcGFnZT9wYXJhbTE9dmFsdWUxJnBhcmFtMj12YWx1ZTI~'",
+    "'text\n that gets converted \\\' into a byte string...'",
+];
+
+pub(crate) const BYTES_FAILS: &[&str] = &[
+    "h64",
+    "b64",
+    "\"\"",
+    "h ''",
+    "b64 ''",
+    "h'001'",
+    "b64'abcdefghijklmnopq #  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~'",
+    "b64'abcdefghijklmnopq   & rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ'",
+    "'\u{7}'",
+];
 
 #[test]
 /// Test if the `HEX_PAIR` rule passes properly.
 fn check_hexpair() {
-    let hex_pairs = vec!["00", "ab", "de", "0f", "f0"];
+    let passes = HEXPAIR_PASSES;
+    let fails = HEXPAIR_FAILS;
 
-    let not_hex_pairs = vec!["0", " 0", "0 ", "az", "0p"];
-
-    for hp in hex_pairs {
-        let parse = CDDLTestParser::parse(Rule::HEX_PAIR, hp);
-        assert!(parse.is_ok());
-    }
-
-    for hp in not_hex_pairs {
-        let parse = CDDLTestParser::parse(Rule::HEX_PAIR, hp);
-        assert!(parse.is_err());
-    }
+    common::check_tests_rule(Rule::HEX_PAIR, passes, fails);
 }
 
 #[test]
 /// Test if the `URL_BASE64` rule passes properly.
 fn check_url_base64() {
-    let tests = vec![
-        "abcdefghijklmnopq   rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~",
-        "abcdefghijklmnopqrstuvwyz0123456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ",
-    ];
-
-    let fails = vec![
-        "abcdefghijklmnopq #  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~ ",
-        "abcdefghijklmnopq $  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\t",
-        "abcdefghijklmnopq %  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\n",
-        "abcdefghijklmnopq ^  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\r",
-        "abcdefghijklmnopq &  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\r\n",
-    ];
+    let passes = URL_BASE64_PASSES;
+    let fails = URL_BASE64_FAILS;
 
-    for test in tests {
-        let parse = CDDLTestParser::parse(Rule::URL_BASE64_TEST, test);
-        assert!(parse.is_ok());
-    }
-
-    for test in fails {
-        let parse = CDDLTestParser::parse(Rule::URL_BASE64_TEST, test);
-        assert!(parse.is_err());
-    }
+    common::check_tests_rule(Rule::URL_BASE64_TEST, passes, fails);
```

Comment:
```suggestion
    common::check_tests_rule(Rule::URL_BASE64_TEST, URL_BASE64_PASSES, URL_BASE64_FAILS);
```

Shouldn't need to assign before use, unless we make an aggregate test case.
This comment applies to all test functions that are structured like this.